src/main/java/io/numaproj/pulsar/consumer/PulsarConsumerManager.java

```diff
@@ -0,0 +1,94 @@
+package io.numaproj.pulsar.consumer;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.client.api.BatchReceivePolicy;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import io.numaproj.pulsar.config.PulsarConsumerProperties;
+
+import javax.annotation.PreDestroy;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * PulsarConsumerManager creates and maintains a single Consumer instance.
+ * A new consumer is created based on the provided batch size and read timeout,
+ * but once created it will be reused until explicitly removed.
+ */
+@Slf4j
+@Component
+@ConditionalOnProperty(prefix = "spring.pulsar.consumer", name = "enabled", havingValue = "true")
+public class PulsarConsumerManager {
+
+    @Autowired
+    private PulsarConsumerProperties pulsarConsumerProperties;
+
+    @Autowired
+    private PulsarClient pulsarClient;
+
+    // The current consumer instance.
+    private Consumer<byte[]> currentConsumer;
+
+    // Returns the current consumer if it exists. If not, creates a new one.
+    public synchronized Consumer<byte[]> getOrCreateConsumer(long count, long timeoutMillis)
```

Comment:
do we need `synchronized`? in numaflow, within a single pod, read is sequentially called.